### PR TITLE
initialize core components

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Initialize core components

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/processingOnlyNode/coreComponents.go [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+	txVersionChecker              process.TxVersionCheckerHandler
+	encodedAddressLen             uint32
+	nodeTypeProvider              core.NodeTypeProviderHandler
+	wasmVMChangeLocker            common.Locker
+	processStatusHandler          common.ProcessStatusHandler
+	hardforkTriggerPubKey         []byte
+	enableEpochsHandler           common.EnableEpochsHandler
+}
+
+type ArgsCoreComponentsHolder struct {
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:76]: missing comments on exported items

File: node/processingOnlyNode/coreComponents.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+		storageFactory.ArgCreatePathManager{
+			WorkingDir: args.WorkingDir,
+			ChainID:    args.Cfg.GeneralSettings.ChainID,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check if we need the real watchdog
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:132]: can remove the TODO, the watchdog should be of type disabled

File: node/processingOnlyNode/coreComponents.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+	instance.watchdog = &watchdog.DisabledWatchdog{}
+	// TODO check if we need the real alarm scheduler
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:134]: can be a disabled component

File: node/processingOnlyNode/coreComponents.go [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+	instance.alarmScheduler = alarm.NewAlarmScheduler()
+	// TODO check if we need the real sync time also this component need to be started
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:136]: can be a disabled component

File: node/processingOnlyNode/coreComponents.go [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+	instance.syncTimer = ntp.NewSyncTime(args.Cfg.NTPConfig, nil)
+	// TODO discuss with Iulian about the round handler
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:138]: yes, here we need a new implementation. The TODO must be left, will be solved in a future PR

File: node/processingOnlyNode/coreComponents.go [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+		EnableEpochsHandler:         instance.enableEpochsHandler,
+	}
+
+	instance.economicsData, err = economics.NewEconomicsData(argsEconomicsHandler)
+	if err != nil {
+		return nil, err
+	}
+	instance.apiEconomicsData = instance.economicsData
+
+	// TODO check if we need this
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:189]: I guess we will need this for the system VM. The TODO can be left, we will decide later.

File: node/processingOnlyNode/coreComponents.go [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+	}
+
+	return uint32(len(encodedEmptyAddress)), nil
+}
+
+func (c *coreComponentsHolder) InternalMarshalizer() marshal.Marshalizer {
+	return c.internalMarshaller
+}
+
+func (c *coreComponentsHolder) SetInternalMarshalizer(marshalizer marshal.Marshalizer) error {
```
Comment (iulianpascalau) [inline node/processingOnlyNode/coreComponents.go:242]: do we need this set?

File: node/processingOnlyNode/coreComponents.go (hunk 1 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:76]: done

File: node/processingOnlyNode/coreComponents.go (hunk 2 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:132]: done

File: node/processingOnlyNode/coreComponents.go (hunk 3 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:134]: done

File: node/processingOnlyNode/coreComponents.go (hunk 4 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:136]: done

File: node/processingOnlyNode/coreComponents.go (hunk 5 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:138]: ok

File: node/processingOnlyNode/coreComponents.go (hunk 7 above)
Comment (miiu96) [reply node/processingOnlyNode/coreComponents.go:242]: no sure I just implemented the methods from the `factory.CoreComponentsHolder`

File: node/processingOnlyNode/coreComponents.go (hunk 1 above)
Comment (iulianpascalau) [reply node/processingOnlyNode/coreComponents.go:78]: not done on all elements

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**